cmd/object-handlers.go

```diff
@@ -720,16 +720,17 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	// Deny if WORM is enabled
-	if globalWORMEnabled {
+	cpSrcDstSame := isStringEqual(pathJoin(srcBucket, srcObject), pathJoin(dstBucket, dstObject))
+
+	// Deny if WORM is enabled. If operation is key rotation of SSE-S3 encrypted object
+	// allow the operation
+	if globalWORMEnabled && !(cpSrcDstSame && crypto.S3.IsRequested(r.Header)) {
```

Comment:
What if source and destination are same and the metadata directive is set to "REPLACE" with WORM enabled? Are we allowing it?

Reply:
I guess not - since that would be a "real" change to the object?!

Reply:
@poornas In the above case, the target metadata gets altered with the "REPLACE" option enabled. If you look here - https://github.com/minio/minio/pull/7429/files#diff-d1c4dccc78d8d9f128ca8972a1199176R949. Such requests should be responded with `ErrMethodNotAllowed`

Reply:
@Praveenrajmani, you are right - updated PR with fix.